packages/@aws-cdk/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -0,0 +1,325 @@
+   *
+   * @default - will use logs.LogGroupProps defaults
+   */
+  readonly logGroupOptions?: logs.LogGroupProps;
+
+  /**
+   * The type of traffic to log. You can log traffic that the resource accepts or rejects, or all traffic.
+   *
+   * @default ALL
+   */
+  readonly trafficType?: FlowLogTrafficType;
+
+  /**
+   * Specifies the type of destination to which the flow log data is to be published.
+   * Flow log data can be published to CloudWatch Logs or Amazon S3
+   *
+   * @default FlowLogDestinationType.toCloudWatchLogs
+   */
+  readonly destination?: FlowLogDestination;
+}
+
+/**
+ * Properties of a VPC Flow Log
+ *
+ * @experimental
+ */
+export interface FlowLogProps extends FlowLogOptions {
+  /**
+   * The name of the FlowLog
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a flowLogName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly flowLogName?: string;
+
+  /**
+   * The type of resource for which to create the flow log
+   */
+  readonly resourceType: FlowLogResourceType;
+}
+
+/**
+ * The base class for a Flow Log
+ *
+ * @experimental
+ */
+abstract class FlowLogBase extends Resource implements IFlowLog {
+  /**
+   * The Id of the VPC Flow Log
+   *
+   * @attribute
+   */
+  public abstract readonly flowLogId: string;
+}
+
+/**
+ * A VPC flow log.
+ * @resource AWS::EC2::FlowLog
+ *
+ * @experimental
+ */
+export class FlowLog extends FlowLogBase {
+  /**
+   * Import a Flow Log by it's Id
+   */
+  public static fromFlowLogId(scope: Construct, id: string, flowLogId: string): IFlowLog {
+    class Import extends FlowLogBase {
+      public flowLogId = flowLogId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The Id of the VPC Flow Log
+   *
+   * @attribute
+   */
+  public readonly flowLogId: string;
+
+  /**
+   * The S3 bucket to publish flow logs to
+   */
+  public readonly bucket?: s3.IBucket;
+
+  /**
+   * The iam role used to publish logs to CloudWatch
+   */
+  public readonly iamRole?: iam.IRole;
+
+  /**
+   * The CloudWatch Logs LogGroup to publish flow logs to
+   */
+  public readonly logGroup?: logs.ILogGroup;
+
+  constructor(scope: Construct, id: string, props: FlowLogProps) {
+    super(scope, id, {
+      physicalName: props.flowLogName
+    });
+
+    const destination = props.destination || new FlowLogDestinationImpl({
+      logDestinationType: 'cloud-watch-logs'
+    });
+
+    const destinationConfig = destination.toDestination();
+
+    if (destinationConfig.logDestinationType === 'cloud-watch-logs') {
+      this.iamRole = destinationConfig.iamRole || new iam.Role(this, 'IAMRole', {
+        roleName: PhysicalName.GENERATE_IF_NEEDED,
+        assumedBy: new iam.ServicePrincipal('vpc-flow-logs.amazonaws.com')
+      });
+
+      this.logGroup = new logs.LogGroup(this, 'LogGroup', props.logGroupOptions);
+
+      new iam.Policy(this, 'Policy', {
```

Comment:
I'd rather see `this.iamRole.addToPolicy()` here to give `iamRole` a chance to NOT have the policy attached (in case it needs to be immutable for locked down environments).

Reply:
updated.